upstash_vector/core/resumable_query.py

```diff
@@ -0,0 +1,100 @@
+from typing import List, Optional, Dict, Any
+from upstash_vector.types import QueryResult
+from upstash_vector.errors import ClientError
+
+RESUMABLE_QUERY_VECTOR_PATH = "/resumable-query"
+RESUMABLE_QUERY_DATA_PATH = "/resumable-query-data"
+RESUMABLE_QUERY_NEXT_PATH = "/resumable-query-next"
+RESUMABLE_QUERY_END_PATH = "/resumable-query-end"
+
+
+class ResumableQuery:
+    def __init__(
+        self, payload: Dict[str, Any], client, namespace: Optional[str] = None
+    ):
+        self.payload = payload
+        self.client = client
+        self.namespace = namespace
+        self.uuid = None
+
+    async def async_start(self) -> List[QueryResult]:
+        if "vector" in self.payload:
+            path = RESUMABLE_QUERY_VECTOR_PATH
+        elif "data" in self.payload:
+            path = RESUMABLE_QUERY_DATA_PATH
+        else:
+            raise ClientError("Payload must contain either 'vector' or 'data' key.")
+
+        if self.namespace:
+            path = f"{path}/{self.namespace}"
+
+        result = await self.client._execute_request_async(
+            payload=self.payload, path=path
+        )
+        self.uuid = result["uuid"]
+        return result["scores"]
+
+    def start(self) -> List[QueryResult]:
```

Comment:
shall we make start private since we made it private in js?

Reply:
We made it private there, because the interface is a bit different. On python sdk we make things a bit more explicitly

Reply:
At the final state, doing this would mess up the interface of the context managers and make the code a bit more complex. So I let it go and made them public. They should be called